### PR TITLE
Further developments for check_grant_data_tools_compatible 

### PR DESCRIPTION
**Pull request by michaelwood**

### CODE REVIEW

**Comment by michaelwood** [comment]
Coverage dropped here a tiny amount due to the conditional on the migration... don't think there is a need for a dedicated test though

File: datastore/db/migrations/0023_auto_20240318_1952.py [hunk 1]
```diff
@@ -4,45 +4,50 @@
...
         ("db", "0022_auto_20240227_1648"),
     ]
 
-    operations = [
-        django.contrib.postgres.operations.BtreeGinExtension(),
-        migrations.AddIndex(
-            model_name="grant",
-            index=django.contrib.postgres.indexes.BTreeIndex(
-                fields=["publisher_org_id"], name="db_grant_publish_53bcd4_btree"
+    if os.environ.get("SKIP_SPECIAL_DB_INDEX"):
```
Comment (R2ZER0) [inline datastore/db/migrations/0023_auto_20240318_1952.py:16]: I'm so used to static languages, I forget you can do magical things like class-level if statements in Python :magic_wand:

File: README.md [hunk 2]
```diff
@@ -13,6 +13,8 @@ $ sudo apt-get install postgresql-12 postgresql-server-dev-12
 $ sudo -u postgres createuser -P -e test  --interactive
 $ createdb -U test -W 360givingdatastore
 
+Note: Special Postgresql extensions are used for indexes, if you don't want to have these installed in the database or the database user doesn't have permissions to install extensions then set environment variable `SKIP_SPECIAL_DB_INDEX=true` before you run migrate.
```
Comment (R2ZER0) [inline README.md]: Does the env var need to be set when running `manage.py test`? If so, mention, otherwise looks good.

**Review by R2ZER0 — DISMISSED** [review]
Just one question, otherwise looks good :+1:

**Review by R2ZER0 — APPROVED** [review]
LGTM :+1:

**Pull request merged**